ckan/lib/helpers.py

```diff
@@ -451,7 +453,7 @@ def _url_for_flask(*args: Any, **kw: Any) -> str:
                 for key, val in kw.items():
                     if isinstance(val, (list, tuple)):
                         for value in val:
-                            if value is None:
+                            if value is None or (isinstance(val, dict) and not val):
```

Comment:
Is there a reason not to
```suggestion
                            if value is None or val == {}:
```

Reply:
@smotornyuk Do you know a workaround for this situation?

Reply:
My bad, I confused the `value` and `val` variables, now it makes sense: [6030628](https://github.com/ckan/ckan/pull/8504/commits/6030628cf2a02d59d897fccddc972852e80cd739)